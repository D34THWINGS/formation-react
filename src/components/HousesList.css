.housesList {
  display: flex;
}

.housesList__house {
  display: flex;
  flex-direction: column;
  align-items: center;
  flex: 0 0 calc(33.33% - 0.33rem);
  padding: 0.5rem;
  background: #333;
  color: #fff;
  box-sizing: border-box;
  border-radius: 5px;
}

.housesList__house:not(:last-child) {
  margin-right: 0.7rem;
}

.house__picture {
  display: block;
  width: 100%;
  border-radius: 5px;
}

.house__name {
  border-bottom: solid 1px #AAA;
  width: 100%;
  padding-bottom: 1rem;
  text-align: center;
  font-weight: bold;
}
