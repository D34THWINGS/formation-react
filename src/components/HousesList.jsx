import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

import houses from '../data/houses';
import StudentsList from './StudentsList';

import gryffindor from '../images/gryffindor.png';
import slytherin from '../images/slytherin.png';
import hufflepuff from '../images/hufflepuff.png';

const images = { gryffindor, slytherin, hufflepuff };

const StyledHousesList = styled.div`
  display: flex;
`;

const House = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  flex: 0 0 calc(33.33% - 0.33rem);
  padding: 0.5rem;
  background: #333;
  color: #fff;
  box-sizing: border-box;
  border-radius: 5px;
  
  &:not(:last-child) {
    margin-right: 0.7rem;
  }
`;

const HousePicture = styled.img`
  display: block;
  width: 100%;
  border-radius: 5px;
`;

const HouseName = styled.p`
  border-bottom: solid 1px #AAA;
  width: 100%;
  padding-bottom: 1rem;
  text-align: center;
  font-weight: bold;
`;

const HousesList = ({ students, onDeleteStudent }) => (
  <StyledHousesList>
    {houses.map(house => (
      <House key={house.id}>
        <HousePicture className="house__picture" src={images[house.picture]} alt={house.name} />
        <HouseName className="house__name">{house.name}</HouseName>
        <StudentsList
          students={students.filter(student => student.house === house.id)}
          onDeleteStudent={onDeleteStudent}
        />
      </House>
    ))}
  </StyledHousesList>
);

HousesList.propTypes = {
  students: PropTypes.arrayOf(PropTypes.shape({
    house: PropTypes.number.isRequired,
  })).isRequired,
};

export default HousesList;
