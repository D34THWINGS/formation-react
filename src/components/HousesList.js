import React from 'react';

import './HousesList.css';
import houses from '../data/houses';
import StudentsList from './StudentsList';

import gryffindor from '../images/gryffindor.png';
import slytherin from '../images/slytherin.png';
import hufflepuff from '../images/hufflepuff.png';

const images = { gryffindor, slytherin, hufflepuff };

const HousesList = ({ students }) => (
  <div className="housesList">
    {houses.map(house => (
      <div className="housesList__house" key={house.id}>
        <img className="house__picture" src={images[house.picture]} alt={house.name} />
        <p className="house__name">{house.name}</p>
        <StudentsList students={students.filter(student => student.house === house.id)} />
      </div>
    ))}
  </div>
);

export default HousesList;
