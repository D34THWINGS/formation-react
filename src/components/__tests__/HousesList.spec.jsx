import React from 'react';
import { shallow } from 'enzyme';

import HousesList from '../HousesList';

describe('HousesList', () => {
  it('should render a list of houses', () => {
    const students = [{ id: 1, house: 2, firstName: 'foo', lastName: 'bar' }];
    const wrapper = shallow(<HousesList students={students} />);
    expect(wrapper).toMatchSnapshot();
  });
});
