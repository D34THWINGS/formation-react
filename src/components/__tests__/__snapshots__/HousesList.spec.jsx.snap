// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HousesList should render a list of houses 1`] = `
<div
  className="housesList"
>
  <div
    className="housesList__house"
    key="1"
  >
    <img
      alt="Gryffindor"
      className="house__picture"
      src="gryffindor.png"
    />
    <p
      className="house__name"
    >
      Gryffindor
    </p>
    <StudentsList
      students={Array []}
    />
  </div>
  <div
    className="housesList__house"
    key="2"
  >
    <img
      alt="Slytherin"
      className="house__picture"
      src="slytherin.png"
    />
    <p
      className="house__name"
    >
      Slytherin
    </p>
    <StudentsList
      students={
        Array [
          Object {
            "firstName": "foo",
            "house": 2,
            "id": 1,
            "lastName": "bar",
          },
        ]
      }
    />
  </div>
  <div
    className="housesList__house"
    key="3"
  >
    <img
      alt="Hufflepuff"
      className="house__picture"
      src="hufflepuff.png"
    />
    <p
      className="house__name"
    >
      Hufflepuff
    </p>
    <StudentsList
      students={Array []}
    />
  </div>
</div>
`;
